### ISSUE TITLE
Inconsistency between PercentageBasedSampler and SamplerProperties

### ISSUE THREAD

**Issue by dhay** [spring-cloud/spring-cloud-sleuth#637]
It looks like there's a discrepancy between the `PercentageBasedSampler` and the `SamplerProperties`. The documentation in `SamplerProperties` states the following:

```
/**
 * Percentage of requests that should be sampled. E.g. 1.0 - 100% requests should be
 * sampled. The precision is whole-numbers only (i.e. there's no support for 0.1% of
 * the traces).
 */
private float percentage = 0.1f;
```
This seems to imply that the configuration value should be specified as a number between 0.0 and 1.0. However, the `PercentageBasedSampler` seems to assume that this value will be a number between 0.0 and 100.0:
```
} else if (this.configuration.getPercentage() == 100) {
```
The documentation on the Spring Cloud Sleuth website seems to support the settings within the `SamplerProperties` class:

> the PercentageBasedSampler is the default if you are using spring-cloud-sleuth-zipkin or spring-cloud-sleuth-stream. You can configure the exports using spring.sleuth.sampler.percentage. The passed value needs to be a double from 0.0 to 1.0 so it’s not a percentage. For backwards compatibility reasons we’re not changing the property name.

Seems like one of these things needs to change, probably the PercentageBasedSampler.

**Comment by marcingrzejszczak** [comment]
Hmm not really. We assume that you have to pass values from `0.0` to `1.0`. The property has `percentage` in it whereas we require passing a double. I don't understand where there's sth inconsistent?

**Comment by dhay** [comment]
The issue is that in the `isSampled` method, there is a check for `configuration.getPercentage() == 100`.  Seems like this check should be `configuration.getPercentage() == 1.0f`

**Comment by marcingrzejszczak** [comment]
Are you sure you're looking at the proper codebase? https://github.com/spring-cloud/spring-cloud-sleuth/blob/master/spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/sampler/PercentageBasedSampler.java#L42

**Comment by marcingrzejszczak** [comment]
It's been fixed on 6th June - https://github.com/spring-cloud/spring-cloud-sleuth/commit/6a0a57ef7fef1f9e85d53fdb03a90fafe1c6e513

**Comment by dhay** [comment]
I was looking at the released 1.2.1 code base.  Good to see that it's been fixed in the most recent code.

**Comment by marcingrzejszczak** [comment]
Thanks for checking it out and keeping an eye on such things!
